src/status_im2/contexts/syncing/scan_sync_code/view.cljs

```diff
@@ -311,17 +311,20 @@
                                       :show-camera?     show-camera?
                                       :content-opacity  content-opacity
                                       :subtitle-opacity subtitle-opacity
-                                      :title-opacity    title-opacity})]
+                                      :title-opacity    title-opacity})
+            view-id                 (rf/sub [:view-id])]
 
         (rn/use-effect
          #(set-listener-torch-off-on-app-inactive torch?))
 
         (when animated?
           (rn/use-effect
            (fn []
-             (rn/hw-back-add-listener reset-animations-fn)
-             #(rn/hw-back-remove-listener reset-animations-fn))
-           [])
+             (when (= view-id :sign-in-intro)
+               (rn/hw-back-add-listener reset-animations-fn))
+             #(when (= view-id :sign-in-intro)
+                (rn/hw-back-remove-listener reset-animations-fn)))
+           [view-id])
```

Comment:
I think there's no need to do double `when`.

```clojure
(rn/use-effect
 (fn []
   (when (= view-id :sign-in-intro)
     (rn/hw-back-add-listener reset-animations-fn)
     #(rn/hw-back-remove-listener reset-animations-fn)))
 [view-id])
```

what about this?  would work?

Reply:
i think cleanup will never get called to remove the back listener as it is defined but never returned from the effect function @ulisesmac

Reply:
I think the cleanup callback is returned by `when` and it should work.
But anyway, it was just a suggestion about code structure.

Reply:
yes, that's correct and thank you for suggestion, resolved